domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1803,50 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// GetAllActivatedModelsUUIDQuery returns a SQL statement that will get all the activated models uuids in the controller.
+func (st *State) GetAllActivatedModelsUUIDQuery() string {
+	return "SELECT uuid from model WHERE activated = true"
+}
+
+// GetActivatedModelUUIDs returns the subset of model uuids from the supplied list that are activated.
+// If no models associated with the uuids are activated then an empty slice is returned.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []coremodel.UUID) ([]coremodel.UUID, error) {
+	if len(uuids) == 0 {
+		return nil, nil
+	}
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type modelUUID struct {
+		UUID coremodel.UUID `db:"uuid"`
+	}
+
+	stmt, err := st.Prepare(`
+SELECT uuid AS &modelUUID.uuid
+FROM   v_model
+WHERE uuid IN ($S[:])
+`, sqlair.S{}, modelUUID{})
+
+	if err != nil {
+		return nil, errors.Errorf("preparing select model activated status statement: %w", err)
+	}
+
+	args := sqlair.S(transform.Slice(uuids, func(s coremodel.UUID) any { return any(s) }))
```

Comment:
`return s` is enough here.

Reply:
If you define a []string type, then no need to have this loop.

Reply:
you still have to accept core types and return them. The loop is unavoidable.

Reply:
Yup previously I was using []string and using custom []string which didnt have to transform the slice, but I think there was a discussion with tom here that we prefer strong types where possible. Would stick with using coremodel.UUID until there are any changes to it in a further discussion. Thanks Kelvin and Tom! :) 

https://github.com/juju/juju/pull/19071#discussion_r1990448572
https://github.com/juju/juju/pull/19071#discussion_r1984383762